prompt2model/model_retriever/generate_hypothetical_document.py

```diff
@@ -427,7 +427,7 @@ def generate_hypothetical_model_description(
     api_call_counter = 0
 
     instruction = prompt.instruction
-    api_agent = APIAgent("gpt-3.5-turbo-16k")
+    api_agent = api_tools.default_api_agent
```

Comment:
Just making sure, this would change the model used for HyDE generation from the 16k context length model to the 4k context length model. Is that intended?

Reply:
It was intentional, sorry I didn't explain. Basically, we'll have to think about how to handle long contexts here. There are two options:

1. Make this model configurable separately from the default, and set it to `gpt-3.5-turbo-16k` in our examples.
2. Make it possible to use HyDe with shorter context windows.

I'd prefer 2. because most models do not support 16k context windows. There are a few recent exceptions, like [Anthropic's Claude](https://www.anthropic.com/index/100k-context-windows), [Adept's new Persimmon model](https://twitter.com/AdeptAILabs/status/1699835884097651145), etc., and it's also possible to extend the context window of various models using [Unlimiformer](https://github.com/abertsch72/unlimiformer), but relying on such long context windows serious limits our modeling choices.

Also RFC: @viswavi and @abertsch72